src/Processors/Transforms/MergeSortingTransform.cpp

```diff
@@ -273,4 +278,11 @@ void MergeSortingTransform::remerge()
     sum_bytes_in_blocks = new_sum_bytes_in_blocks;
 }
 
+ProcessorPtr MergeSortingTransform::getPartialResultProcessor(const ProcessorPtr & current_processor, UInt64 partial_result_limit, UInt64 partial_result_duration_ms)
+{
+    const auto & header = inputs.front().getHeader();
+    auto merge_sorting_processor = std::dynamic_pointer_cast<MergeSortingTransform>(current_processor);
```

Comment:
Do we assure the type? Maybe just some additional check may be intoduced to have a logical error in case of bug.

```suggestion
    auto merge_sorting_processor = std::dynamic_pointer_cast<MergeSortingTransform>(current_processor);
    if (!merge_sorting_processor)
        throw Exception(ErrorCodes:LOGICAL_ERROR, "Unexpected transform type {}", ... );
```

Reply:
@alexX512